### PR TITLE
feat(lang): support const expressions in TS/JS and Dart

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
@tbosch @mprobst is this what you had in mind for `const` support?

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
Putting in M8 as it blocks some issues / PRs in M8

File: modules/angular2/src/facade/lang.dart [hunk 1]
```diff
@@ -10,6 +10,12 @@ class Math {
   static double random() => _random.nextDouble();
 }
 
+// This function is never invoked in Dart, but transformed during TS->Dart
```
Comment (mprobst) [inline modules/angular2/src/facade/lang.dart]: Do we need this at all then? My intention at least was that CONST_EXPR never appears in Dart land.

File: modules/angular2/src/facade/lang.ts [hunk 2]
```diff
@@ -40,6 +40,11 @@ if (assertionsEnabled_) {
 }
 export {int};
 
+// This function is needed only to properly support Dart's const expressions
+// see https://github.com/angular/ts2dart/pull/151 for more info
+export function CONST_EXPR(expr) {
```
Comment (mprobst) [inline modules/angular2/src/facade/lang.ts]: As this only happens in TypeScript, you can make this a generic function:

`export function <T> CONST_EXPR(expr: T) { return expr; }`

Comment (pkozlowski-opensource) [inline modules/angular2/src/facade/lang.ts]: Oh, right - still getting used to the fact that TypeScript is so powerful and supports [generics](http://www.typescriptlang.org/Handbook#generics)

**Comment by pkozlowski-opensource** [comment]
@mprobst updated based on your comments (thnx for the review!), could you please take another look? I would like to land this one ASAP as it is blocking 2 issues for M8

**Comment by mprobst** [comment]
LGTM.

**Pull request closed without merging**